### DISCUSSION TITLE
[WS2122] Homework 4 Discussion

### DISCUSSION THREAD

**Discussion by changkun** [mimuc/gp-ws2021#47] [Discussion] [answered]
This thread is open for the discussion of homework 4.

**Comment by JohannesMerkt** [comment]
![grafik](https://user-images.githubusercontent.com/25324640/145244327-a49fb7df-d4b0-488b-ab97-f9ae26fdb86e.png)
I get this weird result when i try to solve the linear equation with the cholesky solver. Does anyone have an idea why that might be?

**Reply by changkun** [reply]
Looks like an error that boundary vertices are not placed in order.

**Reply by JohannesMerkt** [reply]
now im here
![grafik](https://user-images.githubusercontent.com/25324640/145246150-d60a7dca-2047-442e-b629-78668c93a75b.png)
its probably due to a rounding error

**Reply by changkun** [reply]
That's a good assumption. As we observe from there the solutions are too small. Maybe weights are not properly constructed.

**Reply by JohannesMerkt** [reply]
I cant use the cholesky solver since the weight values are negative.
![grafik](https://user-images.githubusercontent.com/25324640/145252152-5a40b4ae-9df6-44fc-82c6-13f052ba2fac.png)
L*U' = U 
![grafik](https://user-images.githubusercontent.com/25324640/145251812-288cc514-4db2-4c7f-974b-838cf52a041f.png)
I have no idea how i can make them positive but I can use a different solver and that works

**Reply by MarlQ** [reply]
L is not even a lower triangular matrix...
**Edit**: Ok, that is actually irrelevant. I was a bit confused. L needs to be Hermitian positive-definite, which it isn't.

**Reply by MarlQ** [reply]
I tried using a different solver as well, and my solution is suddenly very nice. With Cholesky it doesn't work at all.

**Reply by changkun** [reply]
Oh. Sorry I completely missed the messages here... I hope this does not cause too much trouble.

As discussed in the class, the Cholesky solver is not a suitable solver because the matrix does not satisfy symmetric. For this type of linear equation, we need to use other solvers such as a general LU solver. I should update the comments on the provided code too.

**Comment by alvaxgln** [comment]
When calculating the coordinates for the boundary vertices, they should be evenly distributed on the circle/rectangle, right?
Also can we just pick any radius/side length?

**Reply by JohannesMerkt** [reply]
The uv display area is from negative 0.5 to positive 0.5 in both x and y axis. so if you want the uvs to be in that area you should pick the radius 0.5 and side length 1

**Reply by JohannesMerkt** [reply]
For an optimal result the boundary vertices should be evenly distributed i think

**Reply by alvaxgln** [reply]
Ok, thank you :)

**Reply by changkun** [reply]
Yep, as Johannes mentioned. Uniformly distributed vertices should be better for looping with a fixed step (`for x := vec(0, 0); x.neq(vec(0,0)); x.inc(step) { ... }`)

**Comment by changkun** [comment, marked as answer]
Close due to the end of the submission period.
